src/parser.rs

```diff
@@ -51,6 +51,7 @@ macro_rules! parse_num {
 }
 
 impl ATCParser {
+    #[inline]
```

Comment:
This might be a bit too much, `ATCParser` is not on a hot codepath and I'm not sure if inlining even helps here.

Reply:
If the code path isn't hot, the benefit might be not much significant. I've reverted it.